nodelet/pkg/utils/netutils/net_utils.go

```diff
@@ -79,47 +79,47 @@ func (n *NetImpl) GetNodeIdentifier(cfg config.Config) (string, error) {
 }
 
 // GetNodeIP returns routed network interface IP
-func (n *NetImpl) GetNodeIP() (string, error) {
-	var err error
-	routedInterfaceName, err := n.GetRoutedNetworkInterFace()
-	if err != nil {
-		return "", errors.Wrapf(err, "failed to get routed network interface")
-	}
-	routedIp, err := n.GetIPv4ForInterfaceName(routedInterfaceName)
-	if err != nil {
-		return "", errors.Wrap(err, "failed to get node IP")
+func (n *NetImpl) GetNodeIP(v6Enabled bool) (string, error) {
+	ipFamily := netlink.FAMILY_V4
+	if v6Enabled {
+		ipFamily = netlink.FAMILY_V6
 	}
-	return routedIp, nil
-}
 
-// GetRoutedNetworkInterFace returns roted network interface
-func (n *NetImpl) GetRoutedNetworkInterFace() (string, error) {
-	routedInterface, err := nettest.RoutedInterface("ip", net.FlagUp|net.FlagBroadcast)
+	nodeIp, err := n.GetHostPrimaryIp(ipFamily)
 	if err != nil {
-		return "", err
+		return "", errors.Wrap(err, "failed to get node IP")
 	}
-	routedInterfaceName := routedInterface.Name
-	return routedInterfaceName, nil
+	return nodeIp, nil
 }
 
-// GetIPv4ForInterfaceName returns IPv4 for given interface name
-func (n *NetImpl) GetIPv4ForInterfaceName(interfaceName string) (string, error) {
-	interfaces, _ := net.Interfaces()
-	for _, inter := range interfaces {
-		if inter.Name == interfaceName {
-			addrs, err := inter.Addrs()
-			if err != nil {
-				return "", err
-			}
-			for _, addr := range addrs {
-				switch ip := addr.(type) {
-				case *net.IPNet:
-					if ip.IP.DefaultMask() != nil {
-						return ip.IP.String(), nil
-					}
-				}
+// Gets the host's IP address associated with the default gateway
+func (n *NetImpl) GetHostPrimaryIp(ipFam int) (string, error) {
+	routes, _ := netlink.RouteList(nil, ipFam)
+	for _, route := range routes {
+		if route.Dst != nil {
+			continue
```

Comment:
can you add a comment on what it means when route.Dst is not nil and why we skip such routes?